src/echo_module.cc

```diff
@@ -3,8 +3,8 @@
 
 const std::string EchoModule::typeString = "echo";
 
-Module* EchoModule::createFromParameters(std::shared_ptr<std::map<std::string, std::string>> params) {
-    Module* mod = new EchoModule();
+Module* EchoModule::createFromParameters(std::string path, std::shared_ptr<std::map<std::string, std::string>> params) {
+    Module* mod = new EchoModule(path);
```

Comment:
Hmm, interesting. It's true `path` is inside the map, but if we didn't pass it in, then the callee would need to extract it from the map. I suspect that passing it in will be cheaper than the lookup, and though I guess since it's pass by value it could be an expensive copy. Maybe worth profiling in the future...